### DISCUSSION TITLE
Getting a structured result why a bundle can not be resolved?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-equinox/equinox#452] [Q&A] [answered]
Assume I have a running Equinox framework and there is a bundle that cant be resolved, when I then try to start that bundle (programmatically) it gives a BundleException with some formatted message about why it can't be resolved. 

What I'm looking for is (maybe using internal Equinox API) to get a more structured report than a string I need to parse to pass that into an additional analysis step, is there such method in Equinox?

**Comment by tjwatson** [comment, marked as answer]
See https://github.com/eclipse-equinox/equinox/blob/master/bundles/org.eclipse.osgi/supplement/src/org/eclipse/osgi/report/resolution/ResolutionReport.java

And usage for the OSGi console `diag` command:

https://github.com/eclipse-equinox/equinox/blob/bb979606f8ed98d1641b5e76d62fe222c74fadb8/bundles/org.eclipse.equinox.console/src/org/eclipse/equinox/console/commands/EquinoxCommandProvider.java#L2022-L2034

**Reply by laeubi** [reply]
That looks exactly what I'm looking for!

**Comment by laeubi** [comment]
@tjwatson I see that in case of use-constraint violations one only gets the `ResolutionException`  but felix seems to know a bit more about the possible problems, do you think it would be possible to enhance felix / equinox to have a public accessible `ResolutionException` with additional context? The javadoc of `ResolutionException` explicitly mention the case of extension here so it looks like a good idea.
